manager/apis/app/v1alpha1/m4dapplication_types.go

```diff
@@ -99,6 +99,16 @@ type Condition struct {
 	Message string `json:"message,omitempty"`
 }
 
+// ResourceMetadata contains metadata that allows to query the generated object
+type ResourceMetadata struct {
```

Comment:
Metadata is misleading here. I suggest to use ResourceKey or ResourceReference.

Reply:
Changed to ResourceReference.